src/editor/diff.js

```diff
@@ -57,7 +57,9 @@ export function diffDeletion(oldStr, newStr) {
  * @param {String} oldValue the previous value
  * @param {String} newValue the new value
  * @param {Number} caretPosition the position of the caret after `newValue` was applied.
- * @return {object}
+ * @return {object} an object with `at` as the offset where characters were removed and/or added,
+ *                  `added` with the added string (if any), and
+ *                  `removed` with the removed string (if any)
```

Comment:
we seem inconsistent in our jsdoc: sometimes we put newlines directly against the `*` but other times we indent like this. I'm indifferent, but just something I think we should consider in the future.